HF-24: add stringifyCurrency config callback for TEXT (#1665)
handsontable/hyperformula: docs/guide/known-limitations.md

* The INDEX function doesn't support returning whole rows or columns of the source range – it always returns the contents of a single cell.
* The FILTER function accepts either single rows of equal width or single columns of equal height. In other words, all arrays passed to the FILTER function must have equal dimensions, and at least one of those dimensions must be 1.
* Array-producing functions (e.g., SEQUENCE, FILTER) require their output dimensions to be determinable at parse time. Passing cell references or formulas as dimension arguments (e.g., `=SEQUENCE(A1)`) results in a `#VALUE!` error, because the output size cannot be resolved before evaluation.

> * The TEXT function does not accept embedded double-quote literals in the format string (e.g., `=TEXT(A1, "#,##0.00 ""zł""")` fails at parse time). Use the LCID-tagged form (`[$zł-415] #,##0.00`) or supply a custom [`stringifyCurrency`](configuration-options.md#stringifycurrency) callback that handles such formats outside the parser.

Reviewer: What is the LCID-tag? If I want to display polish currencies in a format "1234,56 zł", can I do it without `stringifyCurrency`?

Author: Good catch — clarified inline in [`known-limitations.md`](https://github.com/handsontable/hyperformula/blob/feature/hf-24-stringify-currency/docs/guide/known-limitations.md): defined the LCID tag as a Microsoft Locale ID with a link to the MS-LCID spec, and explicitly noted that the Polish `"1234,56 zł"` pattern (decimal-comma) requires the `stringifyCurrency` callback because the built-in number formatter always emits `.` as the decimal separator. Fixed in f1eb4efb2.

* The INDEX function doesn't support returning whole rows or columns of the source range – it always returns the contents of a single cell.
* The FILTER function accepts either single rows of equal width or single columns of equal height. In other words, all arrays passed to the FILTER function must have equal dimensions, and at least one of those dimensions must be 1.
* Array-producing functions (e.g., SEQUENCE, FILTER) require their output dimensions to be determinable at parse time. Passing cell references or formulas as dimension arguments (e.g., `=SEQUENCE(A1)`) results in a `#VALUE!` error, because the output size cannot be resolved before evaluation.

> * The TEXT function does not accept embedded double-quote literals in the format string (e.g., `=TEXT(A1, "#,##0.00 ""zł""")` fails at parse time). Use Excel's LCID-tagged form — `[$SYMBOL-LCID]` where LCID is a hex [Microsoft Locale ID](https://learn.microsoft.com/openspecs/windows_protocols/ms-lcid), e.g. `[$zł-415] #,##0.00` for Polish złoty — or supply a custom [`stringifyCurrency`](currency-handling.md) callback that handles such formats outside the parser. For locale-specific patterns like the Polish `"1234,50 zł"` (decimal comma), the callback is required because the built-in number formatter always emits `.` as the decimal separator.

Reviewer: How does Excel handle formats like `=TEXT(A1, "#,##0.00 ""zł""")`?

Author: In Excel, `""` inside a format string is an escape sequence for a literal `"` character — so `#,##0.00 ""zł""` is equivalent to `#,##0.00 "zł"` (a number pattern with `zł` as a quoted literal suffix) and `=TEXT(1234.5, "#,##0.00 ""zł""")` returns `"1,234.50 zł"`. HyperFormula's parser does not support this escape sequence and fails at parse time.

Updated the limitation entry to document what Excel produces before stating the HF restriction, in [8cbb139d4](https://github.com/handsontable/hyperformula/commit/8cbb139d4).

> * The TEXT function does not accept embedded double-quote literals in the format string. In Excel, `""` inside a format string is an escape sequence for a literal `"` character — so `#,##0.00 ""zł""` is equivalent to `#,##0.00 "zł"` (a number pattern with `zł` as a quoted literal suffix) and `=TEXT(1234.5, "#,##0.00 ""zł""")` returns `"1,234.50 zł"`. HyperFormula's parser does not support this escape sequence and fails at parse time. Use Excel's LCID-tagged form — `[$SYMBOL-LCID]` where LCID is a hex [Microsoft Locale ID](https://learn.microsoft.com/openspecs/windows_protocols/ms-lcid), e.g. `[$zł-415] #,##0.00` for Polish złoty — or supply a custom [`stringifyCurrency`](currency-handling.md) callback that handles such formats outside the parser. For locale-specific patterns like the Polish `"1234,50 zł"` (decimal comma), the callback is required because the built-in number formatter always emits `.` as the decimal separator.

Reviewer: ```suggestion
* The TEXT function does not accept embedded double-quote literals in the format string. In Excel, `""` inside a format string is an escape sequence for a literal `"` character — e.g. `=TEXT(1234.5, "#,##0.00 ""zł""")` returns `"1,234.50 zł"`. If your application requires this escape sequence, supply a custom [`stringifyCurrency`](currency-handling.md) callback.
```

I think this is sufficient

Author: Applied in cd024e7a0; the bullet now uses the shorter wording.
